spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/GenerateServerTestsTask.groovy

```diff
@@ -16,99 +16,108 @@
 
 package org.springframework.cloud.contract.verifier.plugin
 
-import groovy.transform.CompileDynamic
 import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.Task
-import org.gradle.api.internal.ConventionTask
+import org.gradle.api.file.Directory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
-
-import org.springframework.cloud.contract.spec.Contract
 import org.springframework.cloud.contract.spec.ContractVerifierException
 import org.springframework.cloud.contract.verifier.TestGenerator
 import org.springframework.cloud.contract.verifier.config.ContractVerifierConfigProperties
 import org.springframework.cloud.contract.verifier.config.TestFramework
-
-import static org.springframework.cloud.contract.verifier.plugin.SpringCloudContractVerifierGradlePlugin.COPY_CONTRACTS_TASK_NAME
+import org.springframework.cloud.contract.verifier.config.TestMode
 
 /**
  * Task used to generate server side tests
  *
  * @since 1.0.0
  */
 @CompileStatic
-class GenerateServerTestsTask extends ConventionTask {
+class GenerateServerTestsTask extends DefaultTask {
+	static final String TASK_NAME = 'generateContractTests'
 
-	File generatedTestSourcesDir
+	@InputDirectory
```

Comment:
If you wanted to make these easier to consume/configure, you can use the `@Nested` qualifer and an object to hold them.

Reply:
Sorry, can you elaborate a bit, pls? How do you think it'll make it easier to configure if I introduce some additional object?

Reply:
So kind of along the same lines as what @marcingrzejszczak was talking about by making a class know about a large number of inputs. The same holds true for tasks. While it's most definitely doable, it blurs the api surface. You can kind of see an example of a way to deal with it here:

https://github.com/shanman190/spring-cloud-contract/commit/34bf9af5eac2822442cb21c90c4960e9d3188d33#diff-b998e42b8cb2cccb4ff11ccc92d5f714

Reply:
Ok, thanks. Will try to clean it up a bit.

Reply:
Cleaned up.
Also tried to switch `ContractVerifierExtension` to be property based, but got some weird issues there and rolled back. Might be a task for some next PR. Not sure will have more time for this now and it's just a nice to have at the end, not a strict requirement.

Reply:
Actually, it looks like I'll manage to use properties in the `ContractVerifierExtension` at the end. Already have something working. Only need to fix tests.